guardian/salesforce-message-handler
# Snyk fix and its collateral

## What does this change?
This change removes aws-lambda-java-log4j v1.0.1 as it contains CVE-2022-23305, with no fix outside of a wholesale update. This change does not update all dependencies, just fixes the vuln. 



## How to test
I reinstated the unit test so as to ensure no runtime errors doing the logging. It required me to default the code to use the DEV environment name, and I had to create a DEV config file in S3 which looked like this:

```
apiClientId = foo
apiToken = bar
salesforceOrganizationId = someOrganizationId
```


## How can we measure success?


I've deployed this in CODE and tested a request via the API Gateway. The CloudWatch logs look identical:

<img width="2016" alt="Screenshot 2023-03-23 at 16 17 25" src="https://user-images.githubusercontent.com/1515970/227267998-148e8231-30a3-4b08-96e9-f46b7343ab8b.png">

## Have we considered potential risks?


No Salesforce updates get synced to Identity or Zuora.
